### PR TITLE
[Shop] Add initial images support

### PR DESCRIPTION
**Pull request by pjedrzejewski**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Related tickets | fixes #X, partially #Y, mentioned in #Z
| License         | MIT

### CODE REVIEW

File: src/Sylius/Component/Core/Model/Product.php [hunk 1]
```diff
@@ -355,4 +355,16 @@ public function getImage()
...
         return $this->getFirstVariant()->getImage();
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getImages()
+    {
+        if (null === $this->getFirstVariant()) {
+            return new ArrayCollection();
```
Comment (CarwynNelson) [inline src/Sylius/Component/Core/Model/Product.php]: I'm not sure if I'm reading this correctly but from the `ProductInterface` it looks like the `firstVariant()` is supposed to return an `ImageInterface` but in the case where there is no first variant we return an `ArrayCollection`. Is this correct? If so is it possible to convert the result of `getImages()` to an `ArrayCollection`, or possibly return a null image interface.

Eg:

```
if (null === $this->getFirstVariant() {
    return new EmptyImageInterface();
}

// etc ...
```

Comment (pjedrzejewski) [inline src/Sylius/Component/Core/Model/Product.php]: @CarwynNelson Actually, it returns first ProductVariant from $variants collection, but product can have no variants at all, if not configured properly yet. Anyway, we will be adding images support on Product, so these methods will return images and first image from product itself. :)

File: src/Sylius/Bundle/UiBundle/Resources/views/Form/theme.html.twig [hunk 2]
```diff
@@ -171,7 +171,7 @@
 
 {% block sylius_image_widget %}
     <div class="ui upload box segment">
-        {% if form.vars.value is null %}
+        {% if form.vars.value.path|default(null) is null %}
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/UiBundle/Resources/views/Form/theme.html.twig]: Do we need `default(null)` here?

**Comment by michalmarcinkowski** [comment]
Thanks Paweł!

**Pull request merged**